core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -424,8 +424,7 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
           if (needsReconfiguration(reconfigurable.reconfigurableConfigs.asJava, updatedMap.keySet)) {
             if (validateOnly)
```

Comment:
Feels like I'm being difficult, but shouldn't the logic be like this?

```scala
             reconfigurable.validateReconfiguration(newConfig)
             if (!validateOnly)
               brokerReconfigurablesToUpdate += reconfigurable
```

Reply:
Sorry, I am running out of excuses! Updated and added unit test.